content/Rust-1.92.0.md

```diff
@@ -0,0 +1,116 @@
+
+However, some functions return `Result`, but the error type they use is not actually "inhabited", meaning it can never exist in real code (e.g. the [`!`](https://doc.rust-lang.org/std/primitive.never.html) or [`Infallible`](https://doc.rust-lang.org/std/convert/enum.Infallible.html) types).
+
+The `unused_must_use` lint now no longer warns on `Result<(), UninhabitedType>`, or on `ControlFlow<UninhabitedType, ()>`. For instance, it will not warn on `Result<(), !>`. This avoids having to check for an error that can never happen.
+
+```rust
+fn can_never_fail() -> Result<(), !> {
+    // ...
+    Ok(())
+}
+
+fn main() {
+    can_never_fail()
+}
+```
+
+This is particularly useful with the common pattern of a trait with an associated error type, where the error type may *sometimes* be infallible:
+
+```rust
+trait UsesAssocErrorType {
+    type Error;
+    fn method(&self) -> Result<(), Self::Error>;
+}
+
+struct CannotFail;
+impl UsesAssocErrorType for CannotFail {
+    type Error = !;
+    fn method(&self) -> Result<(), Self::Error> {
+        Ok(())
+    }
+}
+
+struct CanFail;
+impl UsesAssocErrorType for CanFail {
+    type Error = std::io::Error;
+    fn method(&self) -> Result<(), Self::Error> {
+        Err(std::io::Error::other("something went wrong"))
+    }
+}
+
+fn main() {
+    CannotFail.method(); // No error
+    CanFail.method(); // Error: unused `Result` that must be used
+}
+```
+
+### Emit unwind tables even when `-Cpanic=abort` is enabled on linux
+
+Backtraces with `-Cpanic=abort` previously worked in Rust 1.22 but were broken in Rust 1.23, as we stopped emitting unwind tables with `-Cpanic=abort`. In Rust 1.45 a workaround in the form of `-Cforce-unwind-tables=yes` was stabilized.
+
+In Rust 1.92 unwind tables will be emitted by default even when `-Cpanic=abort` is specified, allowing for backtraces to work properly. If unwind tables are not desired then users should use `-Cforce-unwind-tables=no` to explicitly disable them being emitted.
+
+### Validate input to `#[macro_export]`
+
+Over the past few releases, many changes were made to the way built-in attributes are processed in the compiler. This should greatly improve the error messages and warnings rust gives for built-in attributes and especially make these diagnostics more consistent among all of the over 100 built-in attributes.
+
+To give a small example, in this release specifically, Rust became stricter in checking what arguments are allowed to `macro_export` by [upgrading that check to a "deny-by-default lint" that will be reported in dependencies.](https://github.com/rust-lang/rust/pull/143857). 
+
+### Restrictions on user code impls of `DerefMut` for `Pin`
+
+A soundness issue with `Pin` has been solved by preventing third-party crates from implementing `DerefMut` for `Pin<T>` when `T` is a local type that doesn't implement `DerefMut<Target: Unpin>`.
```

Comment:
This seems a bit hard to grok without context. Maybe at least a PR/issue link would make it clearer?

Reply:
Yeah I think this is probably quite in the weeds for the blog post